src/modules/keyboardmanager/ui/BufferValidationHelpers.cpp

```diff
@@ -205,14 +204,14 @@ namespace BufferValidationHelpers
         if (errorType == KeyboardManagerHelper::ErrorType::NoError)
         {
             KeyShortcutUnion tempShortcut;
-            if (isHybridControl && selectedKeyCodes.size() == 1)
+            if (isHybridControl && std::count_if(selectedCodes.begin(), selectedCodes.end(), [](int32_t a) { return a != -1 && a != 0; }) == 1)
             {
-                tempShortcut = selectedKeyCodes[0];
+                tempShortcut = *std::find_if(selectedCodes.begin(), selectedCodes.end(), [](int32_t a) { return a != -1 && a != 0; });
             }
             else
             {
                 tempShortcut = Shortcut();
-                std::get<Shortcut>(tempShortcut).SetKeyCodes(selectedKeyCodes);
+                std::get<Shortcut>(tempShortcut).SetKeyCodes(selectedCodes);
```

Comment:
Shouldn't we use something similar to `*std::find_if(selectedCodes.begin(), selectedCodes.end(), [](int32_t a) { return a != -1 && a != 0; })` here as well? Since otherwise we may assign `None` as the action key to the shortcut. Earlier we were filtering these

Reply:
I have added check in SetKeyCodes.